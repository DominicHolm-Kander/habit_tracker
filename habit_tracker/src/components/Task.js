import React from 'react'

export default function Task(props) {
    return (
        <div>
            <ul>
                <li>{props.activity.activityName}</li>
                <li>{props.activity.frequency}</li>
                <li>{props.activity.completed ? "Task is complete" : "task outstanding"}</li>
                <li>{props.activity.isPriority ? "This task is priority" : "this is not urgent"}</li>
                <li>{props.activity.lastCompleted}</li>
                <li>{props.activity.streak}</li>
                <li>----------------------</li>
            </ul>
        </div>
    )
}
