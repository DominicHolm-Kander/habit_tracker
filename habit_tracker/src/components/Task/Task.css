/* .completedTask {
    color: green
}

.incompleteTask {
    color: red
}

.completedTask , .incompleteTask { 
    border: 2px solid black
} */