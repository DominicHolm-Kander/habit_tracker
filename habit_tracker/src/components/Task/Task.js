import React from 'react'
import axios from 'axios'
import './Task.css'
import tickImage from '../../pics/tick.png'
import crossImage from '../../pics/cross.png'
import exclamationMarkImage from '../../pics/urgent.png'
import streakFlameImage from '../../pics/streakFlame.png'



export default function Task(props) {
    return (
        <div className={props.className}>
            <h5 className="headbar">Activity Name</h5>
            <h5 className="headbar">Streak</h5>
            <h5 className="headbar">Priority?</h5>
            <h5 className="headbar">Done?</h5>

            <p className="activityName coloumns">{props.activity.activityName}</p>
            <p className="columns">This is the frequency: {props.activity.frequency}</p>
            <p className="columns streak"><img src={streakFlameImage} className="tick" alt="flame"></img>{props.activity.streak}</p>
            <p className="columns">{props.activity.isPriority ? <img src={exclamationMarkImage} className="tick" alt="big red exclamation mark"></img> : null}</p>
            <div className="columns">
            <p className="done">{props.activity.completed ? <img src={tickImage} className="tick" alt="tick"></img> : <img src={crossImage} className="tick" alt="tick"></img>}</p>
            <button className="completedBtn done" type = "button" onClick= {() => axios.post('http://localhost:4000/mark_task_as_complete', {activityName: props.activity.activityName})}>Mark task as complete</button>
            <p className="lastCompleted done">Last completed: {props.activity.lastCompleted.toString().substring(0, 10)}  @  {props.activity.lastCompleted.toString().substring(11, 16)}</p>
            </div>
        </div>
    )
}


