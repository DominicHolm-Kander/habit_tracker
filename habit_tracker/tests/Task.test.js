import React from 'react';
import { shallow } from 'enzyme';
import Task from '../src/components/Task/Task';

describe("shallow Task", () => {
    let wrapper;
    let activity={
        username: 'placeholder',
        activityName: 'placeholder',
        frequency: 'placeholder',
        completed: false,
        isPriority: true,
        lastCompleted: 'placeholder',
        streak: 0
      }

    beforeEach(() => (wrapper = shallow(<Task activity={activity}/>)))
    it("should render correctly", () => {
        expect(wrapper).toMatchSnapshot();
    })
    
})