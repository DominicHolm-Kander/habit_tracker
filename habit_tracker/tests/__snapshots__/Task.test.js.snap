// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow Task should render correctly 1`] = `
<div>
  <ul>
    <li>
      This is the activity name 
      placeholder
    </li>
    <li>
      This is the frequency
      placeholder
    </li>
    <li>
      task outstanding
    </li>
    <li>
      Priority flagger
      This task is priority
    </li>
    <li>
      Last completed (horrible date format)
      placeholder
    </li>
    <li>
      Streak
      0
    </li>
    <button
      onClick={[Function]}
      type="button"
    >
      Mark task as complete
    </button>
  </ul>
</div>
`;
